guides/e2b_autogen/e2b_autogen/main.py

```diff
@@ -0,0 +1,196 @@
+import json
+import logging
+
+from hashlib import md5
+from typing import Optional
+
+# Import E2B
+from e2b import Sandbox
+
+# Import autogen
+from autogen import AssistantAgent, UserProxyAgent, config_list_from_json
+
+# load .env
+from dotenv import load_dotenv
+
+load_dotenv()
+
+# Get your Sandbox session
+sandbox = Sandbox(template="base")
+
+logger = logging.getLogger(__name__)
+
+
+# mock autogen execute_code function with e2b equivalent
+# To run locally: from autogen.code_utils import execute_code
+def execute_code(code: Optional[str] = None,
+                 timeout: Optional[int] = None,
+                 filename: Optional[str] = None,
+                 work_dir: Optional[str] = "/home/user",  # default to e2b default cwd
+                 lang: Optional[str] = "python",
+                 sandbox: Sandbox = None):
+    if all((code is None, filename is None)):
+        error_msg = f"Either {code=} or {filename=} must be provided."
+        logger.error(error_msg)
+        raise AssertionError(error_msg)
+
+    if lang == "node":
+        binary = "node"
+        file_extension = "js"
+    elif lang == "python":
+        binary = "python3"
+        file_extension = "py"
+    else:
+        raise ValueError(f"Unsupported runtime {lang}")
+
+    if filename is None:
+        code_hash = md5(code.encode()).hexdigest()
+        filename = f"{work_dir}/{code_hash}.{file_extension}"
+
+
+    if code is not None:
+        sandbox.filesystem.write(filename, code)
```

Comment:
If code is `None` you won't create the file and the next command will always fail, you should probably raise the error here

Reply:
Maybe it shouldn't be optional 🤔

Reply:
Idea here is that if a filename is passed without code, that local file will be uploaded and executed , that's a gap I can fill or can just simplify per my other comment

Reply:
I'll leave this up to you. But as it's right now, LLM won't just execute already present code, right?